velox/type/TimestampConversion.h

```diff
@@ -107,6 +107,24 @@ Status daysSinceEpochFromWeekDate(
     int32_t dayOfWeek,
     int64_t& out);
 
+/// Computes the (signed) number of days since unix epoch (1970-01-01).
+/// Returns error status if the date is invalid. We treat days of the previous
+/// or next months as a part of the specified WEEK_OF_MONTH. For example, if
+/// weekOfMonth is 5 but the current month only has 4 weeks (such as February),
+/// the first week of March will be considered as the 5th week of February.
```

Comment:
Nice clarification. Perhaps, clarify that the same applies to dayOfWeek. 

BTW, I assume you are matching some behavior from Spark here? It would be nice to clarify that to explain the choice of handling "overflow" dates. Do you have a link to Spark's documentation that explains these rules? Let's make sure we match exactly.

Reply:
Yes, the implemention align with Spark's SimpleDateFormat. I've added docs. You can see the rules in this [link](https://docs.oracle.com/javase/8/docs/api/java/util/Calendar.html).

> Leniency
Calendar has two modes for interpreting the calendar fields, lenient and non-lenient. When a Calendar is in lenient mode, it accepts a wider range of calendar field values than it produces. When a Calendar recomputes calendar field values for return by get(), all of the calendar fields are normalized. For example, a lenient GregorianCalendar interprets MONTH == JANUARY, DAY_OF_MONTH == 32 as February 1.
When a Calendar is in non-lenient mode, it throws an exception if there is any inconsistency in its calendar fields. For example, a GregorianCalendar always produces DAY_OF_MONTH values between 1 and the length of the month. A non-lenient GregorianCalendar throws an exception upon calculating its time or calendar field values if any out-of-range field value has been set.